django/db/models/fields/__init__.py

```diff
@@ -772,6 +769,8 @@ def get_db_prep_lookup(self, lookup_type, value, connection,
             return [self.get_db_prep_value(value, connection=connection,
                                            prepared=prepared)]
         elif lookup_type in ('range', 'in'):
+            if not isinstance(value, (list, tuple)):
```

Comment:
I agree that something needs to change here, or be totally justified why it can't be. I'll spend some more time on this.